### PR TITLE
operate on strings instead of Date objects

### LINKED ISSUE #2022: Pick wrong date in text field
### Steps to reproduce
1. open https://vuetifyjs.com/components/pickers
2. Section `#3 Date pickers - In dialog and menu`
3. Click TextFields
4. Pick a date
5. Get date after 1 month and 1 day on textfield

pick wrong

### Versions

Vuetify: 0.16.0
Browser: Chrome  61.0.3163.100

### Reproduction Link
https://vuetifyjs.com/components/pickers

### LINKED ISSUE #2173: date-picker set v-model in string mode fails
### Steps to reproduce
Open any date-picker with the v-model variable set with string mode and no number days appear in the calendar.

### Versions
v0.16.0-beta.5 and up

### What is expected ?
If the variable is set to '' or '2017-10-31T00:00:00+01:00' the days should appear and in the second case, the date must be selected.

### What is actually happening ?
No number days appear in the calendar and the picker title shows 'Invalid Date' text

### Reproduction Link
https://codepen.io/xevi/pen/KXxZjE

### PR DESCRIPTION
**Pull request by jacekkarczmarczyk**
Besides solving the problem with time zones on Chrome/OSX it also introduces few minor breaking changes:

- removes `dateFormat` and `formattedValue` props - setting `dateFormat` prop only to send a function to Vuetify, call it and get the result back doesn't make sense to me, dev can just run the formatting function by himself on the model
- all input/output values are strings, `value`, `allowedDates` etc must be strings in `YYYY-MM-DD` or `YYYY-MM` format

Fixes #2022 
Fixes #2173

### CODE REVIEW

**Comment by nekosaur** [comment]
Please try to add comments throughout the code. There seems to be quite a bit of code to handle specific Date quirks.

**Comment by jacekkarczmarczyk** [comment]
There'a lot of changed code because it used the Date object thorough the all component and now it uses the string representation. I'll add some comments though, it was needed even before this PR

File: src/components/VDatePicker/mixins/date-table.js [hunk 1]
```diff
@@ -51,9 +38,11 @@ export default {
     },
     dateGenTBody () {
       const children = []
-      const daysInMonth = this.normalizeDate(this.tableYear, this.tableMonth + 1, 0).getDate()
+      const daysInMonth = new Date(this.tableYear, this.tableMonth + 1, 0).getDate()
       let rows = []
-      const day = (this.normalizeDate(this.tableYear, this.tableMonth).getDay() - parseInt(this.firstDayOfWeek) + 7) % 7
+
+      // Use UTC time zone to get the position of the first day of the month relative to the first day of the week
+      const day = (new Date(`${this.tableYear}-${this.tableMonth + 1}-01 GMT+0`).getUTCDay() - parseInt(this.firstDayOfWeek) + 7) % 7
```
Comment (nekosaur) [inline src/components/VDatePicker/mixins/date-table.js]: This isn't very readable. I would suggest breaking it out into a separate function

**Comment by CasperLaiTW** [comment]
Hi @jacekkarczmarczyk 
Is any plan to release about this patch?

**Comment by jacekkarczmarczyk** [comment]
0.17-beta with this patch is coming very soon

**Comment by CasperLaiTW** [comment]
cool, thank you. 👍

**Comment by snevgasymov** [comment]
if I need next format DD/MM/YYY in v-model with a slash, How can I implement it ?

**Comment by jacekkarczmarczyk** [comment]
You can't use formatted value as a model, it accepts only ISO format. You may format it on your own using `momentjs` or similiar libs or your own formatter
For any questions please visit official discord channel https://chat.vuetifyjs.com

**Pull request merged**